agent/cache-types/streaming_health_services.go

```diff
@@ -81,6 +86,11 @@ func (h *healthServicesHandler) State(idx uint64) interface{} {
 	for _, node := range h.state {
 		result.Nodes = append(result.Nodes, node)
 	}
+	result.Nodes.Sort()
 	result.Index = idx
 	return &result
 }
+
+func (h *healthServicesHandler) Reset() {
+	h.state = make(map[string]structs.CheckServiceNode)
```

Comment:
Think it might be worth comment to point out that this isn't actually wiping the result in the cache - until we get a new full snapshot we won't server clients with an empty or partial request but the last good one from before the reset.

That is correct right?

Reply:
Yes that's accurate, do you think we should be resetting the cache entry here too?

Reply:
Nope, keeping "last good result" in cache is important for the "fail static" story. Cache is inherently stale already so an older usable response is better than nothing for this stuff for sure.